data/data-pipeline/data_pipeline/tests/sources/us_army_fuds/test_etl.py

```diff
@@ -0,0 +1,171 @@
+        (-116.10028076199995, 35.83380127000004): "06027000800",
+        (-117.86499023399995, 36.14422607400007): "06027000800",
+        (-155.10320912843935, 19.497857096442765): "15001021010",
+        (-155.91378674587037, 19.516632121497878): "15001021402",
+        (-156.3306524489697, 20.825377142028497): "15009030402",
+        (-156.5429023670438, 20.917074254751412): "15009030800",
+        (-159.48416820625405, 21.907546119100093): "15007040604",
+    }
+    df["GEOID10_TRACT"] = df.geometry.apply(
+        lambda point: lookups[(point.x, point.y)]
+    )
+    return df
+
+
+class TestUSArmyFUDSETL(TestETL):
+    """Tests the FUDS ETL.
+
+    This uses pytest-snapshot.
+    To update individual snapshots: $ poetry run pytest
+            data_pipeline/tests/sources/us_army_fuds/test_etl.py::TestClassNameETL::<testname>
+            --snapshot-update
+    """
+
+    _ETL_CLASS = USArmyFUDS
+
+    _SAMPLE_DATA_PATH = pathlib.Path(__file__).parents[0] / "data"
+    _SAMPLE_DATA_FILE_NAME = "fuds.geojson"
+    _SAMPLE_DATA_ZIP_FILE_NAME = "fuds.geojson"
+    _EXTRACT_TMP_FOLDER_NAME = "USArmyFUDS"
+
+    def setup_method(self, _method, filename=__file__):
+        """Invoke `setup_method` from Parent, but using the current file name.
+
+        This code can be copied identically between all child classes.
+        """
+        super().setup_method(_method=_method, filename=filename)
+
+    def test_init(self, mock_etl, mock_paths):
+        """Tests that the mock NationalRiskIndexETL class instance was
+        initiliazed correctly.
+
+        Validates the following conditions:
+        - self.DATA_PATH points to the "data" folder in the temp directory
+        - self.TMP_PATH points to the "data/tmp" folder in the temp directory
+        - self.INPUT_PATH points to the correct path in the temp directory
+        - self.OUTPUT_PATH points to the correct path in the temp directory
+        """
+        # setup
+        etl = self._ETL_CLASS()
+        # validation
+        assert etl.GEOID_FIELD_NAME == "GEOID10"
+        assert etl.GEOID_TRACT_FIELD_NAME == "GEOID10_TRACT"
+        assert etl.NAME == "us_army_fuds"
+        assert etl.GEO_LEVEL == ValidGeoLevel.CENSUS_TRACT
+        assert etl.COLUMNS_TO_KEEP == [
+            etl.GEOID_TRACT_FIELD_NAME,
+            etl.ELIGIBLE_FUDS_COUNT_FIELD_NAME,
+            etl.INELIGIBLE_FUDS_COUNT_FIELD_NAME,
+            etl.ELIGIBLE_FUDS_BINARY_FIELD_NAME,
+        ]
+
+    def test_get_output_file_path(self, mock_etl, mock_paths):
+        """Tests the right file name is returned."""
+        etl = self._ETL_CLASS()
+        data_path, tmp_path = mock_paths
+
+        output_file_path = etl._get_output_file_path()
+        expected_output_file_path = (
+            data_path / "dataset" / self._ETL_CLASS.NAME / "usa.csv"
+        )
+        assert output_file_path == expected_output_file_path
+
+    def test_fixtures_contain_shared_tract_ids_base(self, mock_etl, mock_paths):
```

Comment:
There are all overridden to be able to mock out `add_tracts_for_geometries`, since loading the tracts over and over is slow, and even if I do some caching it's still slower than we really want in tests.

Reply:
this sounds good to me!